File: Python/228. Summary Ranges.py
class Solution:
    def summaryRanges(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        ans = list()
        l, r = 0, 0
        while l < len(nums):
            r = l
            while r+1 < len(nums) and nums[r+1] == nums[r] + 1:
                r += 1
            if l < r:
                ans.append(str(nums[l]) + "->" + str(nums[r]))
            else:
                ans.append(str(nums[l]))
            l = r+1
        return ans

File: C++/228. Summary Ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> ans;
        for(int i=0;i<nums.size();)
        {
            int l = i,r = i;
            while(r+1<nums.size() && nums[r+1] - 1 == nums[r])
                r+=1;
            if(l<r)
                ans.push_back(to_string(nums[l]) + "->" + to_string(nums[r]));
            else
                ans.push_back(to_string(nums[l]));
            i = r+1;
        }
        return ans;
    }
};